### PR TITLE
Fix support for custom url protocols

### PR DESCRIPTION
**Pull request by alexander-schranz**
| Q | A
| --- | ---
| Bug fix? | no
| New feature? | no
| BC breaks? | no
| Deprecations? | no
| Fixed tickets | part of #5082
| Related issues/PRs | -
| License | MIT
| Documentation PR | https://github.com/sulu/sulu-docs/pull/499

#### What's in this PR?

Fix support for custom url protocols and add `tel:` as default.

#### Why?

We should only validate protocols we know how to validate other protocols should just be ignored as valid. 

#### To Do

- [x] Create a documentation PR
- [ ] Add breaking changes to UPGRADE.md

### CODE REVIEW

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Url/Url.js [hunk 1]
```diff
@@ -22,19 +22,26 @@ type Props = {|
...
 // eslint-disable-next-line max-len
 const URL_REGEX = /^(?:(?:https?|ftps?):\/\/)(?:\S+(?::\S*)?@)?(?:(?!127(?:\.\d{1,3}){3})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]+-?)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,})?(?:\/[^\s]*)?$/iu;
+const URL_PROTOCOLS = ['http://', 'https://', 'ftp://', 'ftps://'];
 
 @observer
 class Url extends React.Component<Props> {
     static defaultProps = {
         disabled: false,
-        protocols: ['http://', 'https://', 'ftp://', 'ftps://'],
+        protocols: [...URL_PROTOCOLS, 'mailto:', 'tel:'],
```
Comment (alexander-schranz) [inline src/Sulu/Bundle/AdminBundle/Resources/js/components/Url/Url.js]: @chirimoya what you think by adding `tel:` to the defaults?

File: src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/Url.js [hunk 2]
```diff
@@ -60,35 +60,34 @@ export default class Url extends React.Component<FieldTypeProps<?string>> {
...
-        }
+        let protocols = undefined;
 
-        const protocols = schemes.map((scheme) => {
-            if (typeof scheme.name !== 'string') {
-                throw new Error(
-                    'Every schema in the "schemes" schemaOption must contain a string string name'
-                );
+        if (schemes) {
+            if (!Array.isArray(schemes) || schemes.length === 0) {
```
Comment (nnatter) [inline src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/Url.js]: we usually use something like `if (defaults !== undefined && (!Array.isArray(schemes) || schemes.length === 0)) {`

Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/containers/Form/fields/Url.js]: the beneath part is not in the this if thats why the nesting is needed

File: src/Sulu/Bundle/AdminBundle/Resources/js/components/Url/Url.js [hunk 3]
```diff
@@ -22,19 +22,26 @@ type Props = {|
...
         valid: true,
     };
 
-    @observable protocol: ?string = undefined;
+    @observable protocol: string;
```
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply src/Sulu/Bundle/AdminBundle/Resources/js/components/Url/Url.js]: changed

File: src/Sulu/Bundle/PageBundle/Content/Types/Url.php [hunk 4]
```diff
@@ -41,6 +41,8 @@ public function getDefaultParams(PropertyInterface $property = null)
                 [
                     'http://' => new PropertyParameter('http://', ''),
                     'https://' => new PropertyParameter('https://', ''),
+                    'mailto:' => new PropertyParameter('mailto:', ''),
```
[the remark this answers is not in the document]
Comment (alexander-schranz) [reply src/Sulu/Bundle/PageBundle/Content/Types/Url.php]: They can be accessed in PHP so I would not remove the DefaultParams when we refractor the content types we should discuss this.

**Pull request merged**